src/dmodule.d

```diff
@@ -261,11 +261,27 @@ public:
 extern (C++) final class Module : Package
 {
 public:
+    alias SetDummy = void[0];
+    // workaround for Issue 15765
+    static struct SymHashWrapper
+    {
+        extern(D) size_t toHash() const @trusted @nogc pure nothrow
+        {
+            return cast(size_t)cast(void*) sym;
+        }
+
+        extern(D) bool opEquals(ref const SymHashWrapper other) const @safe @nogc pure nothrow
+        {
+            return sym is other.sym;
+        }
+        Dsymbol sym;
+        alias sym this;
+    }
     extern (C++) static __gshared Module rootModule;
     extern (C++) static __gshared DsymbolTable modules; // symbol table of all modules
     extern (C++) static __gshared Modules amodules;     // array of all modules
-    extern (C++) static __gshared Dsymbols deferred;    // deferred Dsymbol's needing semantic() run on them
-    extern (C++) static __gshared Dsymbols deferred3;   // deferred Dsymbol's needing semantic3() run on them
+    extern (C++) static __gshared SetDummy[SymHashWrapper] deferred;    // deferred Dsymbol's needing semantic() run on them
+    extern (C++) static __gshared SetDummy[SymHashWrapper] deferred3;   // deferred Dsymbol's needing semantic3() run on them
```

Comment:
We could alternatively move https://github.com/D-Programming-Language/druntime/pull/1282 into the compiler, to avoid using runtime typeinfo.